junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/DefaultDisplayNameGenerator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.descriptor;
+
+import java.lang.reflect.Method;
+
+import org.junit.jupiter.api.extension.DisplayNameGenerator;
+
+/**
+ * Default implementation of the {@link DisplayNameGenerator} API.
+ *
+ * @since 5.4
+ */
+class DefaultDisplayNameGenerator implements DisplayNameGenerator {
+
+	@Override
+	public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+		if (nestedClass.getSimpleName().contains("_")) {
```

Comment:
Needs a better oracle for enabling the "make a sentence" path. Same goes for the analog `if` below in `generateDisplayNameForMethod`.